app/javascript/components/profile/metrics-handler.vue

```diff
@@ -87,47 +88,115 @@ export default {
+        assignmentToResponseTime: this.assignmentToResponseTime[index],
+        responseToApprovalTime: this.responseToApprovalTime[index],
+        approvalToMergeTime: this.approvalToMergeTime[index],
+      }));
+
+      return data;
+    },
+    getTrimmedData() {
+      const trimmedData = this.unfilteredData.filter((currValue) => {
+        if ((currValue.creationToAssignmentTime === 0) && (currValue.assignmentToResponseTime === 0) &&
+          (currValue.responseToApprovalTime === 0) && (currValue.approvalToMergeTime === 0)) {
+          return false;
+        }
+        const validCreationToAssignmentTime = currValue.creationToAssignmentTime <= (
+          this.means.creationToAssignment + this.trimmingFactor * this.standardDeviations.creationToAssignment);
+        const validAssignmentToResponseTime = currValue.assignmentToResponseTime <= (
+          this.means.assignmentToResponse + this.trimmingFactor * this.standardDeviations.assignmentToResponse);
+        const validResponseToApprovalTime = currValue.responseToApprovalTime <= (
+          this.means.responseToApproval + this.trimmingFactor * this.standardDeviations.responseToApproval);
+        const validApprovalToMergeTime = currValue.approvalToMergeTime <= (
+          this.means.approvalToMerge + this.trimmingFactor * this.standardDeviations.approvalToMerge);
+
+        return (validCreationToAssignmentTime && validAssignmentToResponseTime &&
+        validResponseToApprovalTime && validApprovalToMergeTime);
+      });
+
+      return trimmedData;
+    },
+    getDataLabels() {
+      return this.getTrimmedData.map((value) => value.label);
+    },
     prepareDatasets() {
       const datasets = [];
       datasets.push({
         label: this.$i18n.t('message.metrics.creationToAssignmentTimeLabel'),
-        data: this.creationToAssignmentTime,
+        data: this.getTrimmedData.map((value) => value.creationToAssignmentTime),
         color: this.colors.AssignmentTimeColor,
       });
       datasets.push({
         label: this.$i18n.t('message.metrics.assignmentToResponseTimeLabel'),
-        data: this.assignmentToResponseTime,
+        data: this.getTrimmedData.map((value) => value.assignmentToResponseTime),
         color: this.colors.ResponseTimeColor,
       });
       datasets.push({
         label: this.$i18n.t('message.metrics.responseToApprovalTimeLabel'),
-        data: this.responseToApprovalTime,
+        data: this.getTrimmedData.map((value) => value.responseToApprovalTime),
         color: this.colors.ApprovalTimeColor,
       });
       datasets.push({
         label: this.$i18n.t('message.metrics.approvalToMergeTimeLabel'),
-        data: this.approvalToMergeTime,
+        data: this.getTrimmedData.map((value) => value.approvalToMergeTime),
         color: this.colors.MergeTimeColor,
       });
 
       return datasets;
     },
   },
   methods: {
-    getMeanTime(dataArray) {
+    getMean(dataArray) {
```

Comment:
quizás este y el otro (getMean y getStandardDeviation) se podrían llevar a un utils, ya que son cosas como matemáticas. Creo que no tiene tanto sentido que estén dentro del componente como un "this.x". Sí me hace sentido en el mundo del `getMeanTime` porque ahí ya está llevado al dominio de este componente, e incluso quizás (conectándolo con el comentario de antes) hace sentido que sea acá donde se hace la aproximación, y no en un cálculo de promedio "normal".

Sobre el nombre, creo que es el tipo de métodos que deberían ir en el computed, y llamarse sin el get, onda `meanTime`, cosa que después en el template o donde se llamen sea como `this.meanTime` que me hace más sentido semánticamente, no sé que opinai